RTM: Expose PROSPECT coefficients and add `spectra S3` class

ashiklom opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
1. Generalized plate model subroutine underlying PROSPECT has been exposed into its own function (`gpm()`). This function takes one argument -- the wavelength-dependent leaf absorption (`k`) -- and runs the generalized plate model. A related change is that the underlying PROSPECT absorption coefficients are now accessible via `data(dataSpec_prospect<X>`).
2. Added a `spectra` S3 class. This class allows accessing and assigning values by wavelength via `[[` (e.g. `myspec[[400:700]]`), and facilitates plotting (new `plot` and `matplot` methods) spectra against wavelength and combining (`cbind(...)`) multiple spectra by wavelength. 

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
Makes it easier to work with spectra, and perform analyses related to PROSPECT. See `tests/testthat/test.spectra.R` for some examples.

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue) <!-- please add issue number -->
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [X] My change requires a change to the documentation.
- [X] I have updated the CHANGELOG.md.
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

<!--this template is from https://www.talater.com/open-source-templates/#/page/99-->

mdietze commented on CHANGELOG.md:
```diff
@@ -11,6 +11,8 @@ For more information about this file see also [Keep a Changelog](http://keepacha
 
 ### Added
+- RTM: Exposed PROSPECT absorption coefficients and `gpm()` function ("generalized plate model"), facilitating experimentation with different absorption coefficients
+- RTM: Added `spectra` S3 class and methods for subsetting (e.g. `myspec[[400:700]]`), plotting (`plot()` and `matplot()`), and combining spectra by wavelength.
```
These changes seem to rise to the level of needing Documentation updates

serbinsh commented:
@ashiklom do you want to ping us when you want another review of this?

ashiklom commented:
I think I addressed all review comments. @mdietze -- I added a vignette for the new `spectra` class. @dlebauer -- I renamed the R function `gpm` to `generalized_plate_model`. @serbinsh You asked to be pinged as well.

ashiklom commented:
I'm gonna link to this Travis build to show that this does in fact build successfully on Travis for me, just not for PEcAn.

https://travis-ci.org/ashiklom/pecan/builds/332849629

serbinsh commented:
@ashiklom Updated the docs?

ashiklom commented:
@serbinsh @mdietze Here's the update to the vignette (from several weeks ago). Let me know if that looks sufficient for you: https://github.com/PecanProject/pecan/pull/1796/commits/21b1da56fb9f398ee14b4a7f0b9c127d684efd6b

serbinsh commented:
@ashiklom approved, looks like you just need to update the branch

ashiklom commented:
@mdietze This is merged with the latest `develop` and the build is green, so I think this is ready to be merged.
